addons/base_import_module/views/ir_module_views.xml

```diff
@@ -11,7 +11,7 @@
                 </xpath>
                 <xpath expr="//button[@name='button_immediate_install']" position="after">
                     <button type="object" class="btn btn-primary btn-sm" name="button_immediate_install_app" invisible="state != 'uninstalled' or module_type in ('official', False)" groups="base.group_system" context="{'module_name':name}">Activate</button>
-                    <button type="object" class="btn btn-primary btn-sm" name="button_immediate_install_app" invisible="state == 'uninstalled' or module_type in ('official', False)" groups="base.group_system" context="{'module_name':name}">Upgrade</button>
+                    <button type="object" class="btn btn-primary btn-sm" name="button_immediate_upgrade" invisible="state == 'uninstalled' or module_type in ('official', False)" groups="base.group_system" context="{'module_name':name}">Upgrade</button>
```

Comment:
`button_immediate_upgrade` won't do anything here.

What is the issue with demo? It doesn't make sense to include demo data in an upgrade. Maybe we should restrict the upgrade and not allow to check "with demo".

Reply:
yes will revert the change for upgard